src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCaptureCollection.cs

```diff
@@ -116,6 +125,120 @@ internal Capture GetCapture(int i)
 
             return _captures[i];
         }
+
+        int IList<Capture>.IndexOf(Capture item)
+        {
+            var comparer = EqualityComparer<Capture>.Default;
+            for (int i = 0; i < Count; i++)
+            {
+                if (comparer.Equals(this[i], item))
+                    return i;
+            }
+            return -1;
+        }
+
+        void IList<Capture>.Insert(int index, Capture item)
+        {
+            throw new NotSupportedException();
+        }
```

Comment:
We should probably add some exception text to these NotSupportedExceptions. Follow the pattern of what something like ReadOnlyCollection throws.